proto/keyspacepb.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+package keyspacepb;
+
+import "pdpb.proto";
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+// Keyspace provides services to manage keyspaces.
+service Keyspace {
+  rpc UpdateKeyspaceConfig (UpdateKeyspaceConfigRequest) returns (UpdateKeyspaceConfigResponse) {}
+
+  rpc LoadKeyspace (LoadKeyspaceRequest) returns (LoadKeyspaceResponse) {}
+  // WatchKeyspaces first return all current keyspaces' metadata as its first response.
+  // Then, it returns responses containing keyspaces that had their metadata changed.
+  rpc WatchKeyspaces (WatchKeyspacesRequest) returns (stream WatchKeyspacesResponse) {}
+}
+
+message KeyspaceMeta {
+  uint32 id = 1;
+  string name = 2;
+  KeyspaceState state = 3;
+  int64 created_at = 4;
+  int64 state_changed_at = 5;
+  map<string, string> config = 7;
+}
+
+enum KeyspaceState {
+  ENABLED = 0;
+  DISABLED = 1;
+  ARCHIVED = 2;
+}
+
+message UpdateKeyspaceConfigRequest {
+  pdpb.RequestHeader header = 1;
+  string name = 2;
+  // specify config name and values for the target keyspace to update.
+  map<string, string> put = 3;
+  // specify which config to delete, deletion will happen after put operations.
+  repeated string delete = 4;
```

Comment:
Suggest to separate the update and delete interfaces?
If we combine deletion and update in ```UpdateKeyspaceConfigRequest```, users may not pay attention to the internal execution logic. May be results are inconsistent with expectations or users have few questions about the internal implementation. We cannot control the behavior of users only through code comments.

Reply:
I don't think we have to worry too much about this. 
These gRPC interfaces are mostly for horizontal inter-service communications, whoever interacts with it probably already have some knowledge of internal workings (`keyspace ID`, for example). 
For user-facing interface, there's pd-ctl and RESTful API.
Also if we separate update and delete, we lose the ability to update and delete in a single "transaction", causing behavior to deviate from RESTful API's JSON Merge Patch.